numpy/distutils/ccompiler.py

```diff
@@ -191,7 +230,8 @@ def CCompiler_compile(self, sources, output_dir=None, macros=None,
 
     def single_compile(args):
         obj, (src, ext) = args
-        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+        if _needs_build(obj):
+            self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
```

Comment:
Could well be. I need to delete the SciPy build directory in order to run tests that show the problem if it was working in the previous build.